AK/NumberFormat.cpp

```diff
@@ -109,4 +111,31 @@ DeprecatedString human_readable_digital_time(i64 time_in_seconds)
     return builder.to_deprecated_string();
 }
 
+DeprecatedString human_readable_integer(i64 const number)
+{
+    u64 const abs_number = abs(number);
+    // Handle the edge case of number being 0 as log10 is undefined for it.
+    u8 const digit_count = ceil(abs_number > 0 ? log10(static_cast<long double>(abs_number)) : 0) + 1;
+    u8 const section_count = (digit_count - 1) / 3 + 1;
+
+    Vector<u16> separated_sections;
+    separated_sections.ensure_capacity(section_count);
```

Comment:
```suggestion
    Vector<u16,10> separated_sections;
    separated_sections.ensure_capacity(section_count);
```
I feel like adding some inline capacity would still be nice here, `ensure_capacity()` is still doing a heap allocation that could probably be avoided by adding a little inline capacity (since almost all numbers will have < about 10 sections right?)

Reply:
Yeah, absolutely! My bad, I forgot about this.

Reply:
Can you point me to the constructor that I'm supposed to use for this please?
The only one that I see that accepts inline capacity also accepts another vector (line 100 of Vector.h):
```
template<size_t other_inline_capacity>
Vector(Vector<T, other_inline_capacity> const& other)
```

Reply:
You want to use the same constructor than previously (and changes suggested by MacDue should compile as is).
Those are templates parameters of the class, so any class generated from this template will have a constructor that doesn't need to have the parameter passed again.

It's different from templated constructor (like the one you found) which are generated on demand for a specific class(which can be templated or not).

Hope that it's clear 😅, ping me on discord otherwise